### PR TITLE
Add title to gutenberg mobile

### PR DESCRIPTION
**Pull request by diegoreymendez**
## Description

Adds `post-title` component support for mobile.

## How has this been tested?

This PR should be tested through either of these branches:

gutenberg-mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/450
WPiOS: https://github.com/wordpress-mobile/WordPress-iOS/pull/10794

titleinreactnative

## Types of changes

Adds a `.native.js` file with the mobile-specific implementation of `post-title`.

## To do:

Fix the build, which is failing for some reason.  This shouldn't hold the reviews back, though.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/editor/src/components/post-title/index.native.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	handleFocusOutside() {
+		this.onUnselect();
+	}
+
+	onSelect() {
+		this.setState( { isSelected: true } );
+		this.props.clearSelectedBlock();
+	}
+
+	onUnselect() {
```
Comment (pinarol) [inline packages/editor/src/components/post-title/index.native.js]: handleFocusOutside, onSelect and onUnselect functions do not seemed to be called from anywhere, planning to bind these in the future?

File: packages/editor/src/components/post-title/index.native.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		};
+	}
+
+	handleFocusOutside() {
+		this.onUnselect();
+	}
+
+	onSelect() {
+		this.setState( { isSelected: true } );
+		this.props.clearSelectedBlock();
```
Comment (pinarol) [inline packages/editor/src/components/post-title/index.native.js]: I am guessing you are planning to do a separate PR but just wanted to note, clearSelectedBlock() also will need to be propagated to the store.

File: packages/editor/src/components/post-title/index.native.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+	onUnselect() {
+		this.setState( { isSelected: false } );
+	}
+
+	onChange( title ) {
+		this.props.onUpdate( title );
+	}
+
+	onKeyDown( event ) {
```
[the remark this answers is not in the document]
Comment (diegoreymendez) [reply packages/editor/src/components/post-title/index.native.js]: Good point.

After getting to work the de-selection of the currently-selected-block, I'll remove all unused code.

File: packages/editor/src/components/post-title/index.native.js (hunk 1 above)
Comment (diegoreymendez) [reply packages/editor/src/components/post-title/index.native.js]: Just pushed a commit where `onSelect` is wired.

Regarding `onUnselect` it's called from `handleFocusOutside`, which I believe is called by some other editor component - in fact if you check the original (web) .js sources, `handleFocusOutside` is never called.

File: packages/editor/src/components/post-title/index.native.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+const applyWithDispatch = withDispatch( ( dispatch ) => {
+	const {
+		clearSelectedBlock,
+	} = dispatch( 'core/editor' );
+
+	return {
+		clearSelectedBlock,
+	};
```
Comment (pinarol) [inline packages/editor/src/components/post-title/index.native.js]: so this was the thing I was referring to by disptaching to store, just like this one we can also inject onUpdate to props and use editPost( { title } ) to update the post in the store, just like the web side here: https://github.com/WordPress/gutenberg/blob/c2f1059b10ebb3c5457ae61adc05920af4729d58/packages/editor/src/components/post-title/index.js#L180

File: packages/editor/src/components/post-title/index.native.js (hunk 2 above)
Comment (pinarol) [reply packages/editor/src/components/post-title/index.native.js]: thanks! 🎉

**Review by pinarol — APPROVED** [review]
Thanks Diego for this! My tests passed all ok! 🎉 
I think there's one last thing that needs to be addressed and I've added a comment about it but feel free to create an issue for it and handle that separately, at this time it is enough if we just don't forget about it.

**Comment by diegoreymendez** [comment]
Merging, but I'll be going through these PRs tomorrow to list any pending work.

Thanks for the feedback everyone.
